src/lua/core/Import.cpp

```diff
@@ -48,17 +95,267 @@ static std::string get_caller(lua_State *L)
 		if (ar.source[start] == '@') {
 			start++;
 
-			static char source[1024];
-			strncpy(source, &ar.source[start], end - start + 1);
-
-			caller = std::string(source);
+			return std::string(&ar.source[start], end - start + 1);
 		}
 	}
 
 	LUA_DEBUG_END(L, 0);
 	return caller;
 }
 
+/**
+* Generate a lua module name from a given input path.
+* If the path contains a directory component containing a separator (period)
+* this function will return an empty string, as lua module names cannot contain
+* a non-separating period.
+* This function discards the trailing component of the directory path. If you
+* wish to retain it, ensure the path string ends with '/'.
+*/
+static std::string path_to_module(std::string path)
+{
+	std::string module_name;
+	// simple allocation optimizations
+	if (path.size() > 8 && path.find_first_of('/') != std::string::npos) {
+		module_name.reserve(8);
+	}
```

Comment:
I'd be curious to know the story behind this optimisation :)

Reply:
This one comes from the discussion about stdlib containers and your post on optimal sizing for `reserve()` in #4764... however, on second thought the small-string optimization renders this point completely moot. :(